src/Api/ApiServiceProvider.php

```diff
@@ -47,6 +47,14 @@ public function register()
             return $routes;
         });
 
+        $this->app->singleton('flarum.api.csrf-middleware', function () {
+            $checkCsrf = $this->app->make(HttpMiddleware\CheckCsrfToken::class);
+
+            $checkCsrf->ignorePath('/token');
+
+            return $checkCsrf;
+        });
```

Comment:
Now that you are binding those middlewares explicitly per frontend, we could make your code shorter by setting the ignorePath method when the binding is resolved like so:

```php
        $this->app->resolving('flarum.api.csrf-middleware', function (HttpMiddleware\CheckCsrfToken $check) {
            $checkCsrf->ignorePath('/token');
        });
```

Untested +)